Index: LuaGL.c
===================================================================
--- LuaGL.c	(revision 43)
+++ LuaGL.c	(working copy)
@@ -276,14 +276,14 @@
 
    // test arguments
    if(!lua_isstring(L, 1))
-      script_error(L, "incorrect argument to function 'gl.BlendEquation'");
+      luaL_error(L, "incorrect argument to function 'gl.BlendEquation'");
 
    // get values
    mode = (GLenum)get_gl_enum(L, 1);
 
    // test arguments
    if((mode == ENUM_ERROR))
-      script_error(L, "incorrect string argument to function 'gl.BlendFunc'");
+      luaL_error(L, "incorrect string argument to function 'gl.BlendFunc'");
 
    glBlendEquation(mode);
 
@@ -2020,8 +2020,8 @@
       case GL_RED:  case GL_GREEN:  case GL_BLUE:
       case GL_ALPHA: case GL_LUMINANCE: n = 1; break;
       case GL_LUMINANCE_ALPHA:          n = 2; break;
-      case GL_RGB:  case GL_BGR_EXT:    n = 3; break;
-      case GL_RGBA: case GL_BGRA_EXT:   n = 4; break;
+      //case GL_RGB:  case GL_BGR_EXT:    n = 3; break;
+      //case GL_RGBA: case GL_BGRA_EXT:   n = 4; break;
       default:
          ShowError("incorrect string argument to function 'gl.GetTexImage'");
    }
@@ -3625,12 +3625,12 @@
          break;
 
       case GL_RGB:
-      case GL_BGR_EXT:
+      //case GL_BGR_EXT:
          size = 3;
          break;
 
       case GL_RGBA:
-      case GL_BGRA_EXT:
+      //case GL_BGRA_EXT:
          size = 4;
          break;
    }
@@ -3956,7 +3956,7 @@
   {NULL, NULL}
 };
 
-LUAGL_API int luaopen_LuaGL (lua_State *L) {
+LUAGL_API int luaopen_opengl (lua_State *L) {
   luaL_openlib(L, "gl", gllib, 0);
 
   lua_getglobal(L, "gl");
Index: build.sh
===================================================================
--- build.sh	(revision 0)
+++ build.sh	(revision 0)
@@ -0,0 +1,2 @@
+#!/bin/sh
+gcc -I../include/ -I/usr/include/lua5.1/ -llua -lGL -lm -O3 -funroll-loops -fPIC -shared -o opengl.so LuaGL.c LuaGLAux.c
